cli/src/main/java/com/devonfw/tools/ide/tool/graalvm/GraalVm.java

```diff
@@ -0,0 +1,66 @@
+package com.devonfw.tools.ide.tool.graalvm;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link LocalToolCommandlet} for <a href="https://www.graalvm.org/">GraalVM</a>, an advanced JDK with ahead-of-time Native Image compilation.
+ */
+public class GraalVm extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public GraalVm(IdeContext context) {
+
+    super(context, "graalvm", Set.of(Tag.JAVA));
+  }
+
+  @Override
+  public Path getToolPath() {
+
+    return this.context.getSoftwarePath().resolve("extra").resolve(getName());

```

Comment:
Could you add a constant for this folder in `IdeContext` as we had more than one tool using this folder and with string literals we will cause redundancy then.
The general purpose of the `extra` folder is to separate tools that should not be added to the `PATH` variable automatically.